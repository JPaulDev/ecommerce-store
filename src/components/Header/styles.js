import styled from 'styled-components';

export const Header = styled.header`
  background-color: var(--color-grey-1);
  box-shadow: 0 3px 2px 0 rgb(0 0 0 / 20%);
  width: 100%;
`;

export const Container = styled.div`
  align-items: center;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  margin: 0 auto;
  max-width: 1280px;
  padding: 20px 15px;
  width: 97%;
`;

export const HeaderLogo = styled.img`
  height: 36px;
  width: 170px;
`;
