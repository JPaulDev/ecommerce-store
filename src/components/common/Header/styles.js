import styled from 'styled-components';

export const Header = styled.header`
  background-color: var(--color-grey-3);
  box-shadow: var(--box-shadow-lg);
  position: relative;
  width: 100%;
  z-index: 100;
`;

export const Container = styled.div`
  align-items: center;
  display: grid;
  grid-template-columns: 1fr 1fr;
  margin: 0 auto;
  max-width: 1280px;
  padding: 20px 15px;

  @media screen and (min-width: 590px) {
    grid-template-columns: 1fr minmax(max-content, 1fr);
    width: 97%;
  }

  @media screen and (min-width: 935px) {
    grid-template-columns: 1fr minmax(270px, 1fr) minmax(max-content, 1fr);
  }
`;

export const Wrapper = styled.div`
  img {
    display: block;
    height: auto;
    max-width: 100%;
  }

  @media screen and (max-width: 649px) {
    width: 155px;
  }
`;

export const LinksContainer = styled.div`
  column-gap: 20px;
  display: flex;
  justify-content: flex-end;

  @media screen and (min-width: 935px) {
    margin-left: 40px;
  }

  @media screen and (min-width: 1280px) {
    column-gap: 40px;
  }
`;

export const Link = styled.a`
  align-items: center;
  color: var(--color-grey-8);
  column-gap: 10px;
  display: flex;
  font-size: var(--font-size-14);
  position: relative;
  text-decoration: none;
  transition: color 200ms ease-in-out;

  &:hover {
    color: white;
  }
`;

export const TextContainer = styled.div`
  @media screen and (max-width: 589px) {
    display: none;
  }
`;

export const TopText = styled.div`
  color: white;
  font-size: var(--font-size-16);
  line-height: 1;
  margin-bottom: 5px;
`;
