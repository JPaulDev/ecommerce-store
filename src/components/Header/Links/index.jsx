import useMediaQuery from '../../../hooks/useMediaQuery';
import { ReactComponent as AccountIcon } from '../../../assets/svg/account.svg';
import { ReactComponent as BasketIcon } from '../../../assets/svg/basket.svg';
import * as Styled from './styles';

export default function Links() {
  const isMatch = useMediaQuery('(min-width: 575px)');

  return (
    <Styled.Container>
      <Styled.Link href="" aria-label="Account" title="Account">
        <AccountIcon width="32px" height="30px" fill="var(--color-orange-1)" />
        {isMatch && (
          <div>
            <Styled.TopText>Account</Styled.TopText>
            <Styled.Text>Sign in | Register</Styled.Text>
          </div>
        )}
      </Styled.Link>
      <Styled.Link href="" aria-label="Basket" title="Basket">
        <BasketIcon width="34px" height="30px" fill="var(--color-orange-1)" />
        {isMatch && (
          <div>
            <Styled.TopText>Basket</Styled.TopText>
            <Styled.Text>Subtotal Â£0.00</Styled.Text>
          </div>
        )}
      </Styled.Link>
    </Styled.Container>
  );
}
