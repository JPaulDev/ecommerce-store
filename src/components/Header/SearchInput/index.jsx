import { useState, useRef, useId, useEffect } from 'react';
import { ReactComponent as SearchIcon } from '../../../assets/svg/search.svg';
import * as Styled from './styles';

export default function SearchInput() {
  const [query, setQuery] = useState('');
  const [isFocused, setIsFocused] = useState(false);
  const ref = useRef(null);
  const id = useId();

  const handleChange = (e) => setQuery(e.target.value);

  useEffect(() => {
    const element = ref.current;
    const handleFocus = () => setIsFocused(true);
    const handleBlur = () => setIsFocused(false);

    element.addEventListener('focus', handleFocus);
    element.addEventListener('blur', handleBlur);

    return () => {
      element.removeEventListener('focus', handleFocus);
      element.removeEventListener('blur', handleBlur);
    };
  }, []);

  return (
    <Styled.Form
      role="search"
      action="/search"
      method="get"
      isFocused={isFocused}
    >
      <label htmlFor={id} hidden>
        Search
      </label>
      <Styled.Input
        ref={ref}
        id={id}
        value={query}
        name="searchQuery"
        placeholder="Search PC-Connect.co.uk..."
        type="search"
        autoComplete="off"
        maxLength="50"
        required
        onChange={handleChange}
      />
      <Styled.Button title="Search" type="submit">
        <SearchIcon width={30} height={30} fill="white" />
      </Styled.Button>
    </Styled.Form>
  );
}
