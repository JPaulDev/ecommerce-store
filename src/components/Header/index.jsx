import logo from '../../assets/images/header/header-logo.webp';
import SearchInput from './SearchInput';
import Links from './Links';
import * as Styled from './styles';

export default function Header() {
  return (
    <Styled.Header>
      <Styled.Container>
        <Styled.HeaderLogo
          src={logo}
          alt="PC-Connect.co.uk"
          width={153}
          height={32}
        />
        <SearchInput />
        <Links />
      </Styled.Container>
    </Styled.Header>
  );
}
