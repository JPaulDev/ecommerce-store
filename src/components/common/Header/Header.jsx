import Link from 'next/link';
import Image from 'next/future/image';
import logo from '../../../../public/images/header/header-logo.webp';
import { BasketQuantity, BasketSubtotal } from '../../../features/basket';
import { Account, Basket } from '../../icons';
import SearchInput from './SearchInput';
import * as Styled from './styles';

export default function Header() {
  return (
    <Styled.Header>
      <Styled.Container>
        <Styled.Wrapper>
          <Image src={logo} alt="PC-Connect.co.uk" />
        </Styled.Wrapper>
        <SearchInput />
        <Styled.LinksContainer>
          <Link href="/account" passHref>
            <Styled.Link aria-label="Account" title="Account">
              <Account width={32} height={30} />
              <Styled.TextContainer>
                <Styled.TopText>Account</Styled.TopText>
                <div>Sign in | Register</div>
              </Styled.TextContainer>
            </Styled.Link>
          </Link>
          <Link href="/basket" passHref>
            <Styled.Link aria-label="Basket" title="Basket">
              <BasketQuantity />
              <Basket width={34} height={30} />
              <Styled.TextContainer>
                <Styled.TopText>Basket</Styled.TopText>
                <BasketSubtotal />
              </Styled.TextContainer>
            </Styled.Link>
          </Link>
        </Styled.LinksContainer>
      </Styled.Container>
    </Styled.Header>
  );
}
