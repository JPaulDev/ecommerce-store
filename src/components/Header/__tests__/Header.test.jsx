import renderer from 'react-test-renderer';
import Header from '../index';

describe('Header component', () => {
  it('renders a header with an image, search input and links', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
