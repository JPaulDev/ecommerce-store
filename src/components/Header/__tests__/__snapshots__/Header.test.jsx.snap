// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a header with an image, search input and links 1`] = `
<header
  className="sc-iBkjds bVIlMM"
>
  <div
    className="sc-ftvSup kuUWRL"
  >
    <img
      alt="PC-Connect.co.uk"
      className="sc-papXJ hdUloH"
      height={32}
      src="header-logo.webp"
      width={153}
    />
    <form
      action="/search"
      className="sc-bczRLJ bDMbYN"
      method="get"
      role="search"
    >
      <label
        hidden={true}
        htmlFor=":r0:"
      >
        Search
      </label>
      <input
        autoComplete="off"
        className="sc-gsnTZi iltnbf"
        id=":r0:"
        maxLength="50"
        name="searchQuery"
        onChange={[Function]}
        placeholder="Search PC-Connect.co.uk..."
        required={true}
        type="search"
        value=""
      />
      <button
        className="sc-dkzDqf gfrkhO"
        title="Search"
        type="submit"
      >
        <svg
          fill="white"
          height={30}
          width={30}
        >
          search.svg
        </svg>
      </button>
    </form>
    <div
      className="sc-hKMtZM hpnnDO"
    >
      <a
        aria-label="Account"
        className="sc-eCYdqJ edfFZL"
        href=""
        title="Account"
      >
        <svg
          fill="var(--color-orange-1)"
          height={30}
          width={32}
        >
          account.svg
        </svg>
        <div>
          <span
            className="sc-jSMfEi cUKKKr"
          >
            Account
          </span>
          <span
            className="sc-gKXOVf ddmUiw"
          >
            Sign in | Register
          </span>
        </div>
      </a>
      <a
        aria-label="Basket"
        className="sc-eCYdqJ edfFZL"
        href=""
        title="Basket"
      >
        <svg
          fill="var(--color-orange-1)"
          height={30}
          width={34}
        >
          basket.svg
        </svg>
        <div>
          <span
            className="sc-jSMfEi cUKKKr"
          >
            Basket
          </span>
          <span
            className="sc-gKXOVf ddmUiw"
          >
            Subtotal Â£0.00
          </span>
        </div>
      </a>
    </div>
  </div>
</header>
`;
